### PR TITLE
Added av json print

### PR DESCRIPTION
**Pull request by PankajKataria**
added support for printing vtables in JSON format

### CODE REVIEW

**Comment by radare** [comment]
Would be nice to also handle * to print flags for those places. Not just json

Also, do you mind adding support for other archs too? Do this works on x86-32 and x86-64?

File: libr/core/anal.c [hunk 1]
```diff
@@ -3265,18 +3265,25 @@ RList* search_virtual_tables(RCore *core){
...
+		if (printJson) {
+			bool isFirstElement = true;
+			r_cons_printf("[");
+			r_list_foreach (vtables, vtableIter, table) {
+				if (!isFirstElement) r_cons_printf (",");
+				r_cons_printf ("{\"vtable_start_offset\":0x%08"PFMT64x",\"no_of_methods\":%d}",
+				 table->saddr, table->methods);
+				isFirstElement = false;
+			}
+			r_cons_printf("]\n");
```
Comment (radare) [inline libr/core/anal.c]: use r_cons_println, there's no format thing here

File: libr/core/anal.c [hunk 2]
```diff
@@ -3265,18 +3265,25 @@ RList* search_virtual_tables(RCore *core){
...
+	RListIter* vtableIter;
+	vtable_info* table;
+	if (vtables) {
+		if (printJson) {
+			bool isFirstElement = true;
+			r_cons_printf("[");
+			r_list_foreach (vtables, vtableIter, table) {
+				if (!isFirstElement) r_cons_printf (",");
+				r_cons_printf ("{\"vtable_start_offset\":0x%08"PFMT64x",\"no_of_methods\":%d}",
+				 table->saddr, table->methods);
```
Comment (radare) [inline libr/core/anal.c]: dont mix tabs and spaces

File: libr/core/anal.c [hunk 3]
```diff
@@ -3265,18 +3265,25 @@ RList* search_virtual_tables(RCore *core){
...
+	RList* vtables = search_virtual_tables ((RCore *)core);
+	RListIter* vtableIter;
+	vtable_info* table;
+	if (vtables) {
+		if (printJson) {
+			bool isFirstElement = true;
+			r_cons_printf("[");
+			r_list_foreach (vtables, vtableIter, table) {
+				if (!isFirstElement) r_cons_printf (",");
+				r_cons_printf ("{\"vtable_start_offset\":0x%08"PFMT64x",\"no_of_methods\":%d}",
```
Comment (radare) [inline libr/core/anal.c]: this is not valid json. numbers cant be in hexa. use PFMT64d instead without the 0x

Comment (radare) [inline libr/core/anal.c]: no_of_methods and vtable_start_offset looks like pretty bad field names to me. use something like:

{
"offset":38494,
"methods": 33
}

**Comment by radare** [comment]
ping

**Comment by PankajKataria** [comment]
@radare pong :) yes i have made the changes suggested and yes it works on x86_32 and x86_64 currently working on expanding it to multiple arch

File: libr/core/anal.c (hunk 3 above)
Comment (PankajKataria) [inline libr/core/anal.c]: yes i was thinking of that too just made the initial naming according to my convenience

**Comment by PankajKataria** [comment]
@radare  pong something like this for av*

```No   Name                 RealName        Methods    Namehash           Offset     Size       Comment   
==   ====                 ========        ========   ========           ======     ====       ======    
1    obj.vtableforWoman   vtableforWoman  2          0x649e6c7356f43de5 0x00401540 0x00000020 (null)
2    obj.vtableforMan     vtableforMan    2          0x9cbe08191a93701d 0x00401560 0x00000020 (null)
3    obj.vtableforHuman   vtableforHuman  2          0x649e6c73570956a0 0x00401580 0x00000020 (null)
```

**Comment by radare** [comment]
Nope. This output are not r2 commands

The output should be like:


f vtable.Patata=0x738386
f vtable.Human=0x73864
...

**Comment by radare** [comment]
also, do not show "(null)" because asuming printf("%s", NULL); will show that is a bad idea. solaris and hpux libc's will crash

**Comment by radare** [comment]
ping

**Comment by PankajKataria** [comment]
@radare updated

File: libr/core/anal.c [hunk 4]
```diff
@@ -3265,18 +3257,64 @@ RList* search_virtual_tables(RCore *core){
...
+	int wordSize = bits / 8;
+	const char *noMethodName = "No Name found";
+
+	if (vtables) {
+		if (printJson) {
+			bool isFirstElement = true;
+			r_cons_printf("[");
 			r_list_foreach (vtables, vtableIter, table) {
-				printVtable ((RCore *)core, table);
+				if (!isFirstElement) r_cons_printf (",");
```
Comment (radare) [inline libr/core/anal.c]: Missing braces

File: libr/core/anal.c [hunk 5]
```diff
@@ -3265,18 +3257,64 @@ RList* search_virtual_tables(RCore *core){
...
+				r_cons_printf ("{\"offset\":%"PFMT64d",\"methods\":%d}",
+					table->saddr, table->methods);
+				isFirstElement = false;
+			}
+			r_cons_println("]");
```
Comment (radare) [inline libr/core/anal.c]: Missing space

**Comment by radare** [comment]
Merged squashed and fixed those changes in 9fa781beccaca5759b2990cb11358b43c17683ee

thanks

**Pull request closed without merging**